File: Backend-pdf/src/main/java/com/portal/site/services/PostService.java
package com.portal.site.services;

import java.util.List;

import com.portal.site.entities.Post;
import com.portal.site.payloads.PostDto;
import com.portal.site.payloads.PostResponse;

public interface PostService {

	//create 

	PostDto createPost(PostDto postDto,Integer userId);

	//update 

	PostDto updatePost(PostDto postDto, Integer postId);

	// delete

	void deletePost(Integer postId);
	
	//get all posts
	
	PostResponse getAllPost();
	
	//get single post
	
	PostDto getPostById(Integer postId);
	

	

	
	//get all posts by user
	List<PostDto> getPostsByUser(Integer userId);
	
	//search posts
	PostResponse searchPosts(String keyword);

}

File: Frontend-pdf/src/services/post-service.js
import { privateAxios } from "./helper";
import { myAxios } from "./helper";
//create post function
export const createPost = (postData) => {
  //   console.log(postData);
  return privateAxios
    .post(
      `/user/${postData.userId}/posts`,
      postData
    )
    .then((response) => response.data);
};

//get all posts

export const loadAllPosts = (pageNumber, pageSize) => {
  return myAxios
    .get(`/posts`)
    .then((response) => response.data);
};

//load single post of given id
export const loadPost = (postId) => {
  return myAxios.get("/posts/" + postId).then((reponse) => reponse.data);
};

export const createComment = (comment, postId) => {
  return privateAxios.post(`/post/${postId}/comments`, comment);
};

//upload post banner image

export const uploadPostImage = (image, postId) => {
  let formData = new FormData();
  formData.append("pdf", image);
  return privateAxios
    .post(`/post/pdf/upload/${postId}`, formData, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    })
    .then((response) => response.data);
};


export function loadPostUserWise(userId) {
  return privateAxios.get(`/user/${userId}/posts`).then((res) => res.data);
}

//delete post
export function deletePostService(postId) {
  return privateAxios.delete(`/posts/${postId}`).then((res) => res.data);
}

//update post
export function updatePost(post, postId) {
  console.log(post);
  return privateAxios.put(`/posts/${postId}`, post).then((resp) => resp.data);
}


export function searchPosts(text){
  return myAxios.get(`/posts/search/${text}`).then((res) => res.data);
}